### DISCUSSION TITLE
Is ContinueUserActivity feature complete?

### DISCUSSION THREAD

**Discussion by tamingchaos** [shinyorg/shiny#1375] [Q&A]
I've spent a good few hours trying to implement a handler for ContinueUserActivity on iOS; before I spend another few hours, can I confirm if the Shiny implementation here is feature complete? Reviewing the code, It looks like it's almost there, but I've run out of knowledge trying to get anything working in a client application. If this should work, great. I'll spend a little more time and try to get a simple sample together that demonstrates my issue. If it's not complete, I'm happy to take a look at finishing it; I just might need a little help getting going

**Comment by aritchie** [comment]
Do you mean the lifecycle events that Shiny uses?

**Reply by tamingchaos** [reply]
Sorry Allan, 3am Carl wrote the original question, and I understand it lacks any real context. Let me try and correct that.

I'm currently trying to handle an oauth handshake, with the redirectUrl being set to an IOs universal link to handle the call back.
after a few shenanigans configuring the app to handle custom protocols, I think it is now handling them but none of the expected handles are firing as expected.

The reason for the original question was really to ask if anything in this area is missing in Shiny, so I know what side of the fence to dig into next. 

I'm going to try and put a simpler example together to see if i can get it to fire a simple event from a universal link. 

I'm having a few issues debugging with symbols in vscode on Mac, which is a little tiresome, so I might have to sub-module in Shiny src onto my project whilst I try and track this down.

**Reply by aritchie** [reply]
So I did start an event path in the Shiny lifecycle stuff for other Shiny modules, but I have not finished/tested the code that is there for that particular lifecycle event because there isn't any module in Shiny using it yet.
